File: src/main/java/openmedstack/events/DomainEvent.java
package openmedstack.events;

import openmedstack.MessageHeaders;
import openmedstack.events.BaseEvent;

import java.time.Instant;
import java.util.HashMap;
import java.util.concurrent.CompletableFuture;

public abstract class DomainEvent extends BaseEvent {
    private final Integer _version;

    protected DomainEvent(String source, Integer version, Instant timeStamp, String correlationId) {
        super(source, timeStamp, correlationId);
        _version = version;
    }

    public Integer getVersion() {
        return _version;
    }
}

File: modules/openmedstack/src/main/kotlin/org/openmedstack/events/DomainEvent.kt
package org.openmedstack.events

import java.time.OffsetDateTime

abstract class DomainEvent protected constructor(
    private val _source: String,
    val version: Int,
    private val _timeStamp: OffsetDateTime,
    private val _correlationId: String? = null
) : BaseEvent(_source, _timeStamp, _correlationId)